content/tutorial/tutorial.md

```diff
@@ -384,31 +389,31 @@ When we fill the board in later, the `this.state.squares` array will look someth
 ]
 ```
 
-The Board's `renderSquare` method currently looks like this:
+वर्तमान में Board का `renderSquare` मेथड इस तरह दिखता है:
 
 ```javascript
   renderSquare(i) {
     return <Square value={i} />;
   }
 ```
 
-In the beginning, we [passed the `value` prop down](#passing-data-through-props) from the Board to show numbers from 0 to 8 in every Square. In a different previous step, we replaced the numbers with an "X" mark [determined by Square's own state](#making-an-interactive-component). This is why Square currently ignores the `value` prop passed to it by the Board.
+शुरुआत में, हमने प्रत्येक Square में 0 से 8 तक संख्या दिखाने के लिए Board से [`value` prop को नीचे पास](#passing-data-through-props) किया। एक अलग पिछले चरण में, हमने संख्याओं को एक "X" चिह्न के साथ बदल दिया था [वर्ग के अपने state द्वारा निर्धारित](#making-an-interactive-component)। यही कारण है कि वर्तमान में Square ने Board द्वारा इसके लिए दिए गए `value` prop को अनदेखा कर दिया है।
 
-We will now use the prop passing mechanism again. We will modify the Board to instruct each individual Square about its current value (`'X'`, `'O'`, or `null`). We have already defined the `squares` array in the Board's constructor, and we will modify the Board's `renderSquare` method to read from it:
+हम अब props पासिंग मैकेनिज्म का फिर से उपयोग करेंगे। हम प्रत्येक व्यक्तिगत Square को उसके वर्तमान वैल्यू के बारे में निर्देश देने के लिए Board को संशोधित करेंगे(`'X'`, `'O'`, or `null`). हमने पहले से ही Board के constructor में `squares` array को परिभाषित किया है, और हम इसे पढ़ने के लिए Board के `renderSquare` मेथड को संशोधित करेंगे:
 
 ```javascript{2}
   renderSquare(i) {
     return <Square value={this.state.squares[i]} />;
   }
 ```
 
-**[View the full code at this point](https://codepen.io/gaearon/pen/gWWQPY?editors=0010)**
+**[अभी तक का पूरा कोड देखें](https://codepen.io/gaearon/pen/gWWQPY?editors=0010)**
 
-Each Square will now receive a `value` prop that will either be `'X'`, `'O'`, or `null` for empty squares.
+प्रत्येक स्क्वायर को अब एक `value` prop प्राप्त होगा जो या तो खाली वर्गों के लिए `'X'`, `'O'` या `null` होगा।
 
-Next, we need to change what happens when a Square is clicked. The Board component now maintains which squares are filled. We need to create a way for the Square to update the Board's state. Since state is considered to be private to a component that defines it, we cannot update the Board's state directly from Square.
+अगला, हमें यह बदलने की आवश्यकता है कि Square पर क्लिक करने पर क्या होता है। Board कौम्पोनॅन्ट अब रखता है कि कौन से वर्ग भरे हुए हैं। हमें Board की स्थिति को अपडेट करने के लिए Square के लिए एक रास्ता बनाने की आवश्यकता है। चूंकि state को एक ऐसे कौम्पोनॅन्ट के लिए निजी माना जाता है जो इसे परिभाषित करता है, हम Board के state को सीधे Square से अपडेट नहीं कर सकते।
```

Comment:
Board कौम्पोनॅन्ट अब रखता है कि कौन से वर्ग भरे हुए हैं -> revisit

Reply:
एक रास्ता बनाने की आवश्यकता है -> better translation maybe?

Reply:
> Board कौम्पोनॅन्ट अब रखता है कि कौन से वर्ग भरे हुए हैं -> revisit

`Board कौम्पोनॅन्ट अब यह जानकारी रखता है कि कौन से वर्ग भरे हुए हैं।` ?

Reply:
वर्ग  -> square?

Reply:
> एक रास्ता बनाने की आवश्यकता है
I missed this one previously.

`हमें स्क्वायर के लिए एक तरीके की आवशयक्ता है जिससे हम बोर्ड कॉम्पोनेन्ट के स्टेट को अपडेट कर सकें`

How about this?

Reply:
humen ek aise तरीके की आवशयक्ता है jiske dwara Square कौम्पोनॅन्ट Board कौम्पोनॅन्ट के state को अपडेट कर सके